starter-code/ShoppingListWithDetailView/app/src/main/java/ly/generalassemb/drewmahrt/shoppinglistdetailview/DetailActivity.java

```diff
@@ -0,0 +1,40 @@
+package ly.generalassemb.drewmahrt.shoppinglistdetailview;
+
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.widget.TextView;
+
+public class DetailActivity extends AppCompatActivity {
+    public static final String ITEM_ID_KEY = "id";
+    private TextView mNameView;
+    private TextView mDescriptionView;
+    private TextView mPriceView;
+    private TextView mTypeView;
+    private TextView mExpirView;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_detail);
+
+        int itemID = getIntent().getIntExtra(ITEM_ID_KEY, -1);
+        if(itemID==-1){finish();}
```

Comment:
Consider showing the user an error before sending them back to the other activity.